cupy/random/_generator.pyx

```diff
@@ -0,0 +1,226 @@
+
+        Returns an array of samples drawn from the beta distribution. Its
+        probability density function is defined as
+
+        .. math::
+           f(x) = \\frac{x^{\\alpha-1}(1-x)^{\\beta-1}}{B(\\alpha,\\beta)}.
+
+        Args:
+            a (float): Parameter of the beta distribution :math:`\\alpha`.
+            b (float): Parameter of the beta distribution :math:`\\beta`.
+            size (int or tuple of ints): The shape of the array. If ``None``, a
+                zero-dimensional array is generated.
+            dtype: Data type specifier. Only :class:`numpy.float32` and
+                :class:`numpy.float64` types are allowed.
+
+        Returns:
+            cupy.ndarray: Samples drawn from the beta distribution.
+
+        .. seealso::
+            :meth:`numpy.random.Generator.beta
+            <numpy.random.generator.Generator.beta>`
+        """
+        cdef ndarray y
+        y = ndarray(size if size is not None else (), numpy.float64)
+        _launch_dist(self.bit_generator, beta, y, (a, b))
+        return y.astype(dtype)
+
+    def standard_exponential(
+            self, size=None, dtype=numpy.float64,
+            method='inv', out=None):
+        """Standard exponential distribution.
+
+        Returns an array of samples drawn from the standard exponential
+        distribution. Its probability density function is defined as
+
+          .. math::
+             f(x) = e^{-x}.
+
+        Args:
+            size (int or tuple of ints): The shape of the array. If ``None``,
+                a zero-dimensional array is generated.
+            dtype: Data type specifier. Only :class:`numpy.float32` and
+                :class:`numpy.float64` types are allowed.
+            method (str): Method to sample, Currently onlu 'inv', sample from
+                the default inverse CDF is supported.
+            out (cupy.ndarray, optional): If specified, values will be written
+                to this array
+        Returns:
+            cupy.ndarray: Samples drawn from the standard exponential
+                distribution.
+
+        .. seealso:: :meth:`numpy.random.standard_exponential
+                     <numpy.random.mtrand.RandomState.standard_exponential>`
+        """
+        cdef ndarray y
+
+        if method == 'zig':
+            raise NotImplementedError('Ziggurat method is not supported')
+
+        y = ndarray(size if size is not None else (), numpy.float64)
+        _launch_dist(self.bit_generator, exponential, y, ())
+        if out is not None:
+            out[...] = y
+            y = out
+        return y.astype(dtype)
+
+
+def init_curand(generator, state, seed, size):
+    cdef int generator_enum = <int> generator
+    cdef intptr_t state_ptr = <intptr_t>state
+    cdef uint64_t c_seed = <uint64_t>seed
+    cdef uint64_t c_size = <uint64_t>size
+    cdef intptr_t strm = stream.get_current_stream_ptr()
+    init_curand_generator(generator_enum, state_ptr, c_seed, c_size, strm)
```

Comment:
How about the following one for the simplification?
```suggestion
    init_curand_generator(
        <int>generator,
        <intptr_t>state,
        <uint64_t>c_seed,
        <uint64_t>c_size,
        <intptr_t>stream.get_current_stream_ptr()
    )
```

Reply:
Yes, much cleaner thanks!, I (mistakenly) thought what Cython required an explicit declaration with cdef, but seems that this is enough :)